### DISCUSSION TITLE
Version 3

### DISCUSSION THREAD

**Discussion by brutella** [brutella/hkknx-public#314] [Announcements]
Es gibt spannende Neuigkeiten. Die erste [Alpha Version 3.0.0-a1](https://github.com/brutella/hkknx-public/releases/tag/3.0.0-a1) ist da und kann offiziell getestet werden. Nach monatelanger Entwicklung sind nun endlich alle wichtigen Funktionen implementiert. Der aktuelle Stand ist schon sehr vielversprechend. Diese Version habe ich schon selbst einige Wochen produktiv im Einsatz.

**Neue Benutzeroberfläche**
In der Weboberfläche werden in der neuen Version der Status sowie Steuerungselemente für Geräte dargestellt. Somit können die Geräte direkt im Browser gesteuert werden. Die Darstellung der Geräte ist nicht nur für den großen Bildschirm am Computer, sondern auch für mobile Geräte optimiert.

**Monitoring**
Neben der Status-Anzeigen können im Hintergrund auch Monitoring-Daten gesammelt werden. Daten wie Temperatur, Feuchtigkeit und Helligkeit werden dann in Diagrammen dargestellt. Auch die Busauslastung sowie der Systemstatus (CPU, Arbeitsspeicher, Interner Speicher) werden für Monitoringzwecke dargestellt.

(Das Sammeln von Monitoring-Daten muss explizit in den Einstellungen aktiviert werden.)

**Schnellsuche**
Die Schnellsuche über die Weboberfläche ermöglicht das schnelle Finden von Geräten, Skripten und Gruppenadressen.

*Update vom 8. Mai 2024*
Von der aktuellsten Version gibt es eine Demo-Version unter https://hkknx.hochgatterer.me

*Update vom 24. Mai 2024*
**Geräte-Mitteilungen**
Mit Version 3.0.0-a6 können Ereignisse von Geräten (zB Bewegung wurde erkannt, Schloss wurde entriegelt) als Mitteilungen empfangen werden.

*Update vom 11. September 2024*
Version [3.0.0](https://github.com/brutella/hkknx-public/releases/tag/3.0.0) ist jetzt offiziell verfügbar.

[a comment by an automated account is not shown]

**Reply by brutella** [reply]
> Habe auch schon drei kleine Fehler gefunden, und zwar lassen sich Geräte nicht mehr löschen.

Dieses Problem konnte ich nicht reproduzieren. In der Tabellen-Ansicht kann ein Gerät vom (···) Menu aus löschen. Ausserdem kann ein Gerät beim Bearbeiten von den Gruppenadresse gelöscht werden.

> Icon bei Leckagesensor (aktiv) wird nicht gefunden. Wenn Inaktiv ist der Regentropfen zu sehen.

Wird im nächsten Update behoben.

> Aufzeichnung der Wetterdaten funktioniert nicht.

Zurzeit werden nur Daten von HomeKit-Geräten aufgezeichnet. Das wird sich womöglich noch ändern.

**Reply by brutella** [reply]
> Wurde die Übersicht der Inhalte der ETS Datei bewusst weggelassen, oder übersehe ich dies irgendwo?

Ja, die Gruppenadressen sind jetzt viel schneller über die Suche zu finden.

[a comment by an automated account is not shown]

**Comment by brutella** [comment]
Version [3.0.0-a2](https://github.com/brutella/hkknx-public/releases/tag/3.0.0-a2) ist nun verfügbar. Neues in dieser Version ist der vertikale Schieberegler um die Helligkeit von Lampen, Position von Rollläden/Fenstern, und die Lüfterstufen von Lüfter zu verstellen. Die Darstellung dieses Schieberegler ist ähnlich wie in der Apple Home-App.

<img width="322" alt="Screenshot 2024-05-06 at 12 23 49" src="https://github.com/brutella/hkknx-public/assets/125641/56820e8b-0f7f-4f9b-98fc-28f2c26b628d">
<img width="282" alt="Screenshot 2024-05-06 at 12 24 23" src="https://github.com/brutella/hkknx-public/assets/125641/df107990-3402-46cf-af1b-ba0dd1cb3164">

Der Schieberegler unterstützt nicht nur Maus u. Touch Interaktion, sondern auch Pfeiltasten ↑ ↓ in Kombination mit Umschalt-, Kontroll-, und ⌘-Taste.

PS: Die Interaktion mit Rollläden ist somit besser als in der Home-App, da auch ein Stopp-Button zur Verfügung steht. 😉

**Comment by thorsten-gehrig** [comment]
Hallo,
Sieht SUPER aus!
Wo werden die Aufzeichnungsdaten abgelegt? (v.a. für Docker wichtig...)
Gruß
Thorsten

**Reply by brutella** [reply]
Die Daten werden in dem Ordern gespeichert, der mit der Option `--db` beim Start von hkknx angegeben wird.

**Reply by thorsten-gehrig** [reply]
Danke - passt da dieser Ordner natürlich außerhalb des Dockers liegt.
Aber: irgendwie werden bei mir keine Graphen angezeigt - obwohl die Aufzeichnung aktiviert ist...
Irgendeine Idee?

**Reply by brutella** [reply]
Es kann einige Zeit dauern, bis die ersten Daten aufgezeichnet werden. Hängt davon ab, wie oft diese Werte vom KNX Bus empfangen werden.

**Comment by brutella** [comment]
In Version [3.0.0-a5](https://github.com/brutella/hkknx-public/releases/tag/3.0.0-a5) können Geräte über die Weboberfläche gesperrt und entsperrt werden, wenn die Sperren-Eigenschaft konfiguriert wurde. 
<img width="1148" alt="Screenshot 2024-05-15 at 16 59 30" src="https://github.com/brutella/hkknx-public/assets/125641/38af1ce6-976f-4810-af9c-4cb254d0124e">

**Comment by guyyannick** [comment]
Hi,

ich teste auch gerade die 3.0.x und ich muss sagen dass ich ziemlich begeistert bin ! Echt Toller Arbeit 👏🏽

Hier ein Kleiner "Dispay Fehler" bei Temperatur-Angaben mit Nachkommastellen:

<img width="225" alt="image" src="https://github.com/brutella/hkknx-public/assets/63769067/c0947e65-1aa0-4c3c-8a11-acb8f7524ccb">

**Comment by brutella** [comment]
**Update 2024-05-24: Statt Version 3.0.0-a6 unbedingt die Version [3.0.0-a7](https://github.com/brutella/hkknx-public/releases/tag/3.0.0-a7) installieren, da in der alten Version ein fataler Fehler enthalten war.**

Mit Version [3.0.0-a6](https://github.com/brutella/hkknx-public/releases/tag/3.0.0-a6) kann man sich über bestimmte Geräte-Ereignisse durch Mitteilungen informieren lassen. Zum Beispiel, wenn ein Bewegungsmelder eine Bewegung erkennt, oder wenn ein Schloss entriegelt wird.

In den Geräte-Einstellungen wird festgelegt, bei welchen Ereignissen an welche Geräte ein Mitteilung versendet werden soll.

---

Ausserdem können Mitteilungen jetzt auch an Browser gesendet werden. Dazu wird ein Browser in den Einstellungen unter Mitteilungen registriert. Versendet werden die Nachrichten an die Browser über die [Push API](https://developer.mozilla.org/en-US/docs/Web/API/Push_API?retiredLocale=de).

Alle gängigen Browser unterstützen diese Schnittstelle. Erfolgreich testen konnte ich diese Funktion bereits mit Safari unter macOS 13. Leider gibt es unter iOS 17.4 Probleme aufgrund von dem [DMA](https://developer.apple.com/support/dma-and-apps-in-the-eu).

Mit Chrome unter Android sollte es auch funktionieren. Leider besitze ich kein Android und konnte diese Funktion daher nicht testen. Wäre toll, wenn sich jemand zum Testen findet.

**Comment by brutella** [comment]
Mit Version [3.0.0-b2](https://github.com/brutella/hkknx-public/releases/tag/3.0.0-b2) werden Monitoring-Daten automatisch aggregiert um weniger Speicherplatz zu benötigen. Dieser Vorgang wird automatisch ausgeführt.

In diesem Schritt wurde auch die Datenbank-Implementierung geändert. Bestehende Daten müssen daher migriert werden. Dazu gibt es unter Einstellungen → Monitoring-Daten die Funktion "Migrate Data".

**Comment by flobxy** [comment]
Werden die Monitoring Daten in der InfluxDB gespeichert die in v2 integriert wurde? Falls ja, ist es möglich sich mit der InfluxDB zu verbinden um weitere Daten (z.B. Überschuss Photovoltaik die ich via Skript in eine InfluxDB schiebe) visualiseren zu können?

**Reply by brutella** [reply]
> Werden die Monitoring Daten in der InfluxDB gespeichert

Nein

**Reply by flobxy** [reply]
Die seit v2 integrierte InfluxDB, ist es möglich sich mit dieser zu verbinden?

**Reply by brutella** [reply]
hkknx hat keine InfluxDB integriert. Es ist jedoch mit dem [influxdb-Package](https://hochgatterer.me/de/hkknx/scripts/influxdb/) möglich, über ein Skript Daten in eine InfluxDB zu speichern.

**Reply by flobxy** [reply]
Danke, stimmt, so war das. InfluxDB musste ich mir selbst installieren.

**Comment by brutella** [comment]
In Version [3.0.0-b4](https://github.com/brutella/hkknx-public/releases/tag/3.0.0-b4) werden Verlaufsdaten nicht nur von konfigurierten Geräten angezeigt werden, sondern auch von einer beliebigen Gruppenadresse.

Das Diagramm wird in der Detailansicht von einer Gruppenadresse in der Suche angezeigt.
Man gibt entweder den Namen oder die Adresse der Gruppenadresse ein und wählt die gewünschte Adresse aus.

In der Detailansicht werden der letzte empfangene Wert und die Verlaufsdaten angezeigt.
<img width="807" alt="Screenshot 2024-06-26 at 14 04 36" src="https://github.com/brutella/hkknx-public/assets/125641/09d52c24-c6bf-43ec-ba01-8eda062ceddb">

Somit bekommt man einen Überblick über Sensorenwerte, die nicht von HomeKit unterstützt werden zB Stromverbrauch.

**Wichtig:** Die Daten können nur angezeigt werden, wenn ein Datentyp im ETS Projekt für die Gruppenadresse angegeben wurde. Ansonsten können die empfangenen Bytes nicht verarbeitet werden.

**Reply by flobxy** [reply]
Sehr cool. Das sieht toll aus!!

![IMG_5120](https://github.com/brutella/hkknx-public/assets/1721741/860d3ffd-8ca8-4ff3-8b3c-83b96c513374)

**Reply by vinc32** [reply]
Wie hast du denn den Energiezähler eingebunden? Bin auf [v3.0.0-b9.1] und wüsste nicht wie ich das mache.

**Reply by flobxy** [reply]
> Wie hast du denn den Energiezähler eingebunden? Bin auf [v3.0.0-b9.1] und wüsste nicht wie ich das mache.

Im ETS konfiguriert. Projekt in hkknx aktualisiert und nach dem Namen gesucht. Aber du hast Recht. So wie damals im Juni klappt es nicht mehr. Ich bin auch auf 9.1

**Reply by vinc32** [reply]
Das Problem ist ja eher - dass es keinen Energiezähler zum auswählen als Gerät gibt. In iOS18 soll das ja funktionieren - aber ggf. wurd es von @brutella wieder vorerst rausgenommen?

**Reply by flobxy** [reply]
Im HomeKit, korrekt. Im hkknx sollte es dennoch klappen, wenn du in der Suche die Adresse findest. Bin aber gerade auch mobil unterwegs. Der Screeni war vom Desktop. Ich probiere es nachher nochmals.

**Reply by flobxy** [reply]
Ich habe es gerade mal schnell auf dem Desktop ausprobiert. Via Suche finde ich anscheinend keine "nicht HomeKit" Geräte (wie den Energiezähler). Also den Screeni von oben kann ich nicht mehr in 9.1 reproduzieren.

**Reply by vinc32** [reply]
ok vielen dank, dachte schon ich stelle mich an

**Reply by brutella** [reply]
Wenn du den Namen oder die Adresse  in die Suche eingibst, sollte die Gruppenadresse als Ergebnis angezeigt werden.

**Reply by vinc32** [reply]
Reboot hat geholfen ;) Danke

**Comment by brutella** [comment]
In Version [3.0.0-b5](https://github.com/brutella/hkknx-public/releases/tag/3.0.0-b5) wird nicht nur der letzte Wert einer Gruppenadresse angezeigt, sondern die 10 Werte innerhalb der letzten 7 Tage.

Demo → https://hkknx.hochgatterer.me/?sgaid=cpuhaf9q6bodsbntao10&sq=ga%3A0%2F0%2F10

<img width="802" alt="Screenshot 2024-06-28 at 10 38 35" src="https://github.com/brutella/hkknx-public/assets/125641/d77e246a-8fc5-44a3-a94d-9a69e02d9f21">

**Reply by vinc32** [reply]
In deiner Demo hast du in der Seitenleiste die Werte für Temperatur, Lüftung und co dargestellt, dafür fehlen Sie in den Feldern der Räume. Wie genau wurde das realisiert? Über das anpassen der Felder finde ich es nicht, und auch sonst wüsste ich nicht wo man diese Setting konfigurieren kann.

**Reply by brutella** [reply]
> dafür fehlen Sie in den Feldern der Räume

Die Werte sollten in den Feldern auftauchen, wenn du die Kategorie in der Seitenleiste anklickst.

Siehe hier → https://hkknx.hochgatterer.me/projects/cotlava698vmorb3og3g/accessories?ctype=climate&dt=tiles

<img width="1463" alt="Screenshot 2024-08-27 at 16 14 59" src="https://github.com/user-attachments/assets/895e4b1e-0a5d-47a7-aeff-1ab17725068a">

**Reply by vinc32** [reply]
> wenn du die Kategorie in der Seitenleiste anklickst.

<img width="1983" alt="Bildschirmfoto 2024-08-28 um 22 46 43" src="https://github.com/user-attachments/assets/8fb3d3fe-4915-422f-97cc-4a4de2967872">
EDIT: tauchen auf ;) Die schönen Zusammenfassungen von Temperatur etc in der Seitenleiste fehlt aber trotzdem ;)

**Comment by vinc32** [comment]
[v3.0.0-b9.2]
Der Status aus KNX wird nicht Zuverlässig gelesen. Sobald eine Lampe via KNX geschalten wird aktualisiert sich der Status nicht zuverlässig in HKKNX/Home.
Ich kann kein Muster erkennen wann es nicht aktualisiert.

**Reply by brutella** [reply]
Kannst du bitte mal einen Screenshot von der Konfiguration der Lampe hier posten?
Wie hoch ist deine Buslast? (Siehe Monitoring → Busauslastung)

**Reply by vinc32** [reply]
Die Buslast ist bei 0-4%
<img width="652" alt="Bildschirmfoto 2024-08-27 um 09 50 01" src="https://github.com/user-attachments/assets/144f149c-3ab9-45ea-84fd-cd7cb3c8debf">

Die Konfig einer Beispiellampe
<img width="1322" alt="Bildschirmfoto 2024-08-27 um 09 52 31" src="https://github.com/user-attachments/assets/7986938d-3a73-4b47-bcf6-34664e35559f">

2.
<img width="1298" alt="Bildschirmfoto 2024-08-27 um 09 53 07" src="https://github.com/user-attachments/assets/7f4bf4de-d956-4e74-ada9-648c8f56c1ac">

**Reply by brutella** [reply]
Wenn du das Licht für "Essen" über KNX schaltest, empfängst du dann unter Monitoring → Gruppenmonitor einen Wert von der Gruppenadresse `2/1/36`?

**Reply by vinc32** [reply]
nein - aber in der ETS schon - das ist kurios. Der Schalten Wert wird aber erkannt. Ich starte mal den KNX IP Router neu vllt hat er irgendwas

**Comment by brutella** [comment]
Mit Version [3.0.0-b9.3](https://github.com/brutella/hkknx-public/releases/tag/3.0.0-b9.3) wurde die Darstellung auf Smartphones optimiert. Auch das Design der Navigationsbar hat sich etwas geändert. Bin auf das Feedback gespannt.

Das neue Design kann schonmal in der [Online-Demo](https://hkknx.hochgatterer.me) angesehen werden.

**Reply by vinc32** [reply]
Sieht gut aus - man kann das Monitoring jetzt auf dem iPhone richtig öffnen und auf alle Daten zugreifen und man muss nicht mehr an den Rechner um fix zu schauen ob alles passt.
